src/components/Match/TeamfightMap/index.jsx

```diff
@@ -0,0 +1,174 @@
+import React, { Component } from 'react';
+import { formatSeconds, calculateDistance, calculateRelativeXY } from 'utility';
+import ReactTooltip from 'react-tooltip';
+import Measure from 'react-measure';
+import { IconRadiant, IconDire } from 'components/Icons';
+import Table from 'components/Table';
+import { teamfightColumns } from 'components/Match/matchColumns';
+import styles from './index.css';
+
+const MAP_WIDTH = 400;
+const iconSize = mapWidth => (mapWidth / 12 <= 15 ? 15 : mapWidth / 12);
+
+const style = (width, position) => ({
+  width: iconSize(width),
+  height: iconSize(width),
+  top: ((width / 127) * position.y) - (iconSize(width) / 2),
+  left: ((width / 127) * position.x) - (iconSize(width) / 2),
+});
+
+const isRadiant = radiantGoldDelta => radiantGoldDelta > 0;
+
+const TeamfightIcon = ({ isRadiant, position, tooltipKey, mapWidth = MAP_WIDTH }) => {
+  const IconType = isRadiant ? IconRadiant : IconDire;
+
+  return (
+    <IconType
+      className={styles.teamfightIcon}
+      style={style(mapWidth, position)}
+      data-tip
+      data-for={tooltipKey}
+    />
+  );
+};
+
+const getIconStyle = radiantGoldDelta => (isRadiant(radiantGoldDelta) ? styles.radiant : styles.dire);
+
+const Teamfight = ({ position, tooltipKey, start, end, radiantGoldDelta, selected, mapWidth }) => (
+  <div>
+    <div
+      className={`${selected && styles.selected} ${getIconStyle(radiantGoldDelta)}`}
+    >
+      <TeamfightIcon
+        position={position}
+        isRadiant={isRadiant(radiantGoldDelta)}
+        tooltipKey={tooltipKey}
+        mapWidth={mapWidth}
+      />
+    </div>
+    <ReactTooltip
+      id={tooltipKey}
+      effect="solid"
+    >
+      <div>{formatSeconds(start)} - {formatSeconds(end)}</div>
+      <div>
+        <span className={styles.goldChange}>
+          {isRadiant(radiantGoldDelta) ? radiantGoldDelta : radiantGoldDelta * -1}
+        </span> Gold
+      </div>
+    </ReactTooltip>
+  </div>
+);
+
+const avgPosition = ({ deaths_pos: deathPositions }) => {
```

Comment:
It's just an average of all the position data, so it could be off if the teamfight was that long. I was thinking about enhancing this in the next iteration to show a little tombstone or something at each death when selected, so that would sort of solve that issue.